scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -305,15 +338,17 @@ object Execution {
     type Counters = Map[Long, ExecutionCounters]
     private[this] val cache = new FutureCache[(Config, Execution[Any]), (Any, Counters)]
     private[this] val toWriteCache = new FutureCache[(Config, ToWrite), Counters]
+    protected[EvalCache] val filesToCleanup = mutable.Set[String]()
```

Comment:
actually, I think we should keep this private and override the add method. The reason is we forgot one more thing: this is not thread-safe, but we have so many threads in play, we should make this thread-safe.

I think we are modifying this value from several threads.

So, if we just forward the method, we can handle the locking a bit more cleanly inside that method.